openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,850 @@
-                delete scaledSviIndicator[k];
-            }
-
-            for (var k in weightedSviIndicator) {
-                delete weightedSviIndicator[k];
-            }
-
-            function newSVIvalues(SVIaverage, tempMunic) {
-                sviIndicator[tempMunic] = SVIaverage;
+                var munic4 = la[o].properties[selectedRegion];
+                var theme4 = name;
+                tempString.push(munic4 + '|'+ theme4 +'|'+ tempSum);
             }
+        }
+    }
 
-            // Associate the newSVIvalues method with the svi object
-            sviIndicator.SVIvalues = newSVIvalues;
-            var tempSviParentChildKey = [];
-            var tempSviIndicator = {};
-            for (var i = 0; i < municipality.length; i++) {
-                tempSviIndicator[municipality[i]] = [];
-            }
-            tempSviSearchElements = Object.keys(weightedCatIndicator);
-            tempSviSearchElements.pop();
-            var sessionKey = Object.keys(weightedCatIndicator[tempSviSearchElements[0]]);
-            sessionKey.pop();
-            sessionKey.pop();
+    for (var p5 = 0; p5 < tempString.length; p5++) {
+        // capture an array for each record
+        var temp;
+        temp = tempString[p5];
+        temp = temp.split("|");
+        generateObject(temp);
+    }
 
-            for (var i = 0; i < tempSviSearchElements.length; i++) {
-                var tempArray = [];
+    Category_PCP_Chart(catData);
 
-                // Grab the municipality of the iteration
-                var tempMunic = weightedCatIndicator[i]["municipality"];
+    /////////////////////////
+    //// Compute the SVI ////
+    /////////////////////////
 
-                // Remove the municipality from the sessionkey array
-                var index = sessionKey.indexOf("municipality");
+    var SVI = {};
+    var sviNameLookUp = 'SVI';
+    var sviJSONthemes = socialVulnIndex;
+    // SVI is an object with region and value
+    SVI = combineIndicators(sviNameLookUp, catData, sviJSONthemes );
 
-                if (index > -1) {
-                    sessionKey.splice(index, 1);
-                }
+    ///////////////
+    //// Scale ////
+    ///////////////
 
-                // Build the temp object
-                for (var j = 0; j < sessionKey.length; j++) {
-                    tempArray.push(weightedCatIndicator[i][sessionKey[j]]);
-                }
+    scale(SVI);
 
-                var SVIsum = 0
-                $.each(tempArray,function() {
-                    SVIsum += this;
-                });
+    ////////////////////////////////
+    //// Compute the risk index ////
+    ////////////////////////////////
 
-                sviIndicator.SVIvalues(SVIsum, tempMunic);
-            };
+    createIndexSimple(la, riskIndex);
+    var riskIndicator = createIndex(la, riskIndex);
 
+    // capture all risk indicators for selection menu
+    for (var key in riskIndicator[0]) {
```

Comment:
A better structured riskIndicators object would let you avoid this `if`, because you would be browsing homogeneous data